### ISSUE TITLE
Convert Trained Torch Model to Caffe model?

### ISSUE THREAD

**Issue by 3DTOPO** [jcjohnson/fast-neural-style#127]
Hello,

I am trying to convert a trained torch model to caffe with Facebook's torch2caffe utility based on fblualib. 

But I get the error: `unknown Torch class <nn.ShaveImage>`

Any ideas how to fix, or how to generate a caffe model instead of a Torch model or other ways of converting to caffe?

Here is the full output from running the torch2caffe command:
`ubuntu@ip-172-31-25-94:~/fb-caffe-exts2$ th torch2caffe/torch2caffe.lua --input ~/the_wave.t7 --prototxt the_wave.prototxt --caffemodel the_wave.the_wave.prototxt
Opts: %s	{
  prototxt = "the_wave.prototxt",
  format = "lua",
  input = "/home/ubuntu/the_wave.t7",
  caffemodel = "the_wave.the_wave.prototxt",
  preprocessing = "",
  input_dims = {
    0
  },
  verify = "",
  input_tensor = ""
}	
/home/ubuntu/torch/install/bin/luajit: /home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:343: unknown Torch class <nn.ShaveImage>
stack traceback:
	[C]: in function 'error'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:343: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:409: in function 'load'
	./torch2caffe/lib.lua:185: in function 'main'
	torch2caffe/torch2caffe.lua:23: in main chunk
	[C]: in function 'dofile'
	...untu/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670`

**Comment by htoyryla** [comment]
The Torch model is a serialized torch object, which contains references to Torch classes, both predefined and user defined (like nn.ShaveImage which is not a standard Torch class by defined in fast-neural-style code).

Caffe, as far as I know, supports only those layers which are predefined in caffe. This excludes most of the recent torch based models, as which use custom layers. I am not aware of any simple way to solve this (other than implementing each custom class into caffe).

**Comment by lzx1413** [comment]
@3DTOPO you can avoid using shaveimage layer with using zero padding and replace instance norm with mvn layer or batch norm layer if the batchsize is one
